pkg/server/plugin/zmq/handler.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2017-present Oursky Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zmq
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/skygeario/skygear-server/pkg/server/router"
+)
+
+const ZMQWorkerIDContextKey string = "ZMQWorkerIDContextKey"
+
+type zmqResponseWriter struct {
+	header int
+	response []byte
+}
+func (w *zmqResponseWriter) Header() http.Header {
+	return map[string][]string{}
+}
+func (w *zmqResponseWriter) Write(body []byte) (int, error) {
+	w.response = append(w.response, body...)
+	return len(body), nil
+}
+func (w *zmqResponseWriter) WriteHeader(status int) {
+	w.header = status
+}
+
+func (p *zmqTransport) SetRouter(router *router.Router) {
+	p.router = router
+}
+
+func (p *zmqTransport) handleRequest(workerID string, buffer []byte, responseChan chan []byte) {
+	reader := bytes.NewReader(buffer)
+	data := map[string]interface{}{}
+	if jsonErr := json.NewDecoder(reader).Decode(&data); jsonErr != nil && jsonErr != io.EOF {
+		// TODO(b123400): handle json error
+		return
+	}
+
+	payload := &router.Payload{
+		Context: context.Background(),
+		Meta: map[string]interface{}{},
+		Data: data,
+		AccessKey: router.MasterAccessKey,
+	}
+
+	responseWriter := &zmqResponseWriter{}
+	resp := router.NewResponse(responseWriter)
+
+	newCtx := context.WithValue(payload.Context, ZMQWorkerIDContextKey, workerID)
+	payload.Context = newCtx
+
+	// TODO(b123400), put real path and method here
+	p.router.HandlePayload("schema/fetch", "POST", payload, resp)
```

Comment:
The current implementation depends on both path and method, so I have to pass it in.
We either:
- Make it path / method independent, a payload alone is enough for routing
- Hide the path / method somewhere in the payload, maybe `Meta` as you said